airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.kt

```diff
@@ -0,0 +1,281 @@
+import io.airbyte.cdk.integrations.base.AirbyteMessageConsumer
+import io.airbyte.cdk.integrations.base.Destination
+import io.airbyte.cdk.integrations.base.IntegrationRunner
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.base.SerializedAirbyteMessageConsumer
+import io.airbyte.cdk.integrations.base.TypingAndDedupingFlag.getRawNamespaceOverride
+import io.airbyte.cdk.integrations.base.adaptive.AdaptiveDestinationRunner
+import io.airbyte.cdk.integrations.destination.NamingConventionTransformer
+import io.airbyte.cdk.integrations.destination.jdbc.JdbcCheckOperations.attemptTableOperations
+import io.airbyte.cdk.integrations.destination.staging.StagingConsumerFactory.Companion.builder
+import io.airbyte.integrations.base.destination.typing_deduping.CatalogParser
+import io.airbyte.integrations.base.destination.typing_deduping.DefaultTyperDeduper
+import io.airbyte.integrations.base.destination.typing_deduping.NoOpTyperDeduperWithV1V2Migrations
+import io.airbyte.integrations.base.destination.typing_deduping.ParsedCatalog
+import io.airbyte.integrations.base.destination.typing_deduping.TyperDeduper
+import io.airbyte.integrations.base.destination.typing_deduping.migrators.Migration
+import io.airbyte.integrations.destination.snowflake.migrations.SnowflakeState
+import io.airbyte.integrations.destination.snowflake.typing_deduping.SnowflakeDestinationHandler
+import io.airbyte.integrations.destination.snowflake.typing_deduping.SnowflakeSqlGenerator
+import io.airbyte.integrations.destination.snowflake.typing_deduping.SnowflakeV1V2Migrator
+import io.airbyte.integrations.destination.snowflake.typing_deduping.SnowflakeV2TableMigrator
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog
+import java.util.*
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.function.Consumer
+import javax.sql.DataSource
+import net.snowflake.client.core.SFSession
+import net.snowflake.client.core.SFStatement
+import net.snowflake.client.jdbc.SnowflakeSQLException
+import org.apache.commons.lang3.StringUtils
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class SnowflakeDestination
+@JvmOverloads
+constructor(
+    private val airbyteEnvironment: String,
+    private val nameTransformer: NamingConventionTransformer = SnowflakeSQLNameTransformer(),
+) : BaseConnector(), Destination {
+
+    override fun check(config: JsonNode): AirbyteConnectionStatus? {
+        val snowflakeInternalStagingSqlOperations =
+            SnowflakeInternalStagingSqlOperations(nameTransformer)
+        val dataSource = getDataSource(config)
+        try {
+            val database = getDatabase(dataSource)
+            val outputSchema = nameTransformer.getIdentifier(config["schema"].asText())
+            attemptTableOperations(
```

Comment:
mega nit: I'd prefer this to not be statically imported, so it's easier to see that it's defined in the Jdbc class